### PR TITLE
wasm: add fail_open for telemetry plugins

### PR DESCRIPTION
**Pull request by kyessenov**
Exposes the [fail_open](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/wasm/v3/wasm.proto#extensions-wasm-v3-pluginconfig) flag from Envoy, removing the need to use EnvoyFilter for Wasm filters.

### CODE REVIEW

File: extensions/v1alpha1/wasm.proto [hunk 1]
```diff
@@ -311,6 +311,12 @@ message WasmPlugin {
   // namespace of the `WasmPlugins`. Defaults to `0`.
   google.protobuf.Int32Value priority = 10;
 
+  // Enables the fail open behavior for the Wasm plugin fatal errors to bypass
+  // the plugin execution. A fatal error can be a failure to fetch the remote
+  // binary, an exception, or abort() on the VM. This flag is not recommended
+  // for the authentication or the authorization plugins.
+  bool fail_open = 13;
```
Comment (howardjohn) [inline extensions/v1alpha1/wasm.proto:318]: is there any possibility we will want other values in the future that may point to an enum? Like

`STRICT|ALLOW_FETCH_FAILURE|ALLOW_ANY_FAILURE` or something possibly, if we distinguish from exception vs fetching failing?

**Review by howardjohn — COMMENTED** [review]
Overall seems reasonable, just one minor syntax question

File: extensions/v1alpha1/wasm.proto (hunk 1 above)
Comment (kyessenov) [reply extensions/v1alpha1/wasm.proto:318]: I don't think we need fine-grained fail-open modes in the medium term. For security filters, nothing should fail. For telemetry filters, there's no need to treat some failures as special.

Comment (kyessenov) [reply extensions/v1alpha1/wasm.proto]: Makes sense, changed to enum.

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**